cogent/src/Cogent/PrettyPrint.hs

```diff
@@ -680,10 +680,16 @@ instance Pretty DataLayoutSize where
   pretty (Bytes b) = literal (string (show b ++ "B"))
   pretty (Add a b) = pretty a <+> symbol "+" <+> pretty b
 
+instance Pretty Endianness where
+  pretty LE = keyword "LE"
+  pretty BE = keyword "BE"
+  pretty ME = keyword "ME"
+
 instance Pretty d => Pretty (DataLayoutExpr' d) where
   pretty (RepRef n s) = if null s then reprname n else parens $ reprname n <+> hsep (fmap pretty s)
   pretty (Prim sz) = pretty sz
   pretty (Offset e s) = pretty e <+> keyword "at" <+> pretty s
+  pretty (Endian e n) = pretty e <+> keyword "using" <+> pretty n
```

Comment:
if `e` is `ME`, it will print `using ME`, which I don't think is valid syntax according to your parser.